src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -366,101 +344,121 @@ protected void appendFailurePrefix(StringBuilder failureMessage, String netID, i
-     * is modified in the future to perform actual updates.
-     */
     protected String updateACHAccountIfNecessary(Person payee, PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
         StringBuilder processingResults = new StringBuilder();
-        processingResults.append("Update functionality NOT implemented yet. Payee \"")
-            .append(achDetail.getNetID()).append("\" data being checked for possible mismatch. ");
-
-        if (!StringUtils.equals(achDetail.getBankRoutingNumber(), achAccount.getBankRoutingNumber())
-                || !StringUtils.equals(achDetail.getBankAccountNumber(), achAccount.getBankAccountNumber())) {
-            /*
-             * For a future enhancement, we will modify this method to actually update the Payee ACH Account
-             * via a maintenance document, and possibly use more fields for change comparison.
-             */
-            processingResults.append(" Account information from input file for payee type '")
-                .append(achAccount.getPayeeIdentifierTypeCode())
-                .append("' does not match what is in KFS, but will be left as-is.");
+
+        if (accountHasChanged(achDetail, achAccount)) {
+            PayeeACHData achData = new PayeeACHData(payee, achDetail, achAccount.getPayeeIdentifierTypeCode(), achAccount);
+            String accountUpdateErrors = createAndRoutePayeeACHAccountDocument(achData, this::setupDocumentForACHUpdate);
+            processingResults.append(accountUpdateErrors);
         } else {
-            processingResults.append(" Input file's account information for payee of type '")
-                .append(achAccount.getPayeeIdentifierTypeCode())
-                .append("' matches what is already in KFS; no updates will be made for this entry.");
+            LOG.info("Input file's account information for payee of type '" + achAccount.getPayeeIdentifierTypeCode()
+                    + "' matches what is already in KFS; no updates will be made for this entry.");
+        }
+        
+        if (processingResults.length() > 0) {
+            processingResults.append(" Update was NOT performed.");
+            LOG.warn("updateACHAccountIfNecessary: " + processingResults.toString());
         }
-        processingResults.append(" Update was NOT performed.");
-        LOG.warn("updateACHAccountIfNecessary: " + processingResults.toString());
+        
         return processingResults.toString();
     }
 
+    protected boolean accountHasChanged(PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
+        return !StringUtils.equals(achDetail.getBankRoutingNumber(), achAccount.getBankRoutingNumber())
+                || !StringUtils.equals(achDetail.getBankAccountNumber(), achAccount.getBankAccountNumber())
+                || !StringUtils.equals(
+                        getACHTransactionCode(achDetail.getBankAccountType()), achAccount.getBankAccountTypeCode());

```

Comment:
I don't think an equalsIgnoreCase is strictly necessary for the bank routing or account numbers, but it might be worth doing for the account type code. I will update some or all of these to equalsIgnoreCase checks accordingly.

Reply:
After further review and consideration, I don't think any of these particular checks need to be changed to equalsIgnoreCase. The account type codes should only be one of the codes from our ACH Transaction Code BOs. I also had an email discussion with Marcia a few months ago concerning this; she said the bank routing number should always be numeric, and to the best of her knowledge US bank account numbers should also be numeric (and we are only using the ACH feature for US bank accounts). Marcia did say that foreign bank accounts can be alphanumeric, but we're not using ACH for those.

Reply:
Thank you for confirming.